lib/Parse/ParseDecl.cpp

```diff
@@ -3349,6 +3352,8 @@ ParserResult<PoundDiagnosticDecl> Parser::parseDeclPoundDiagnostic() {
 }
 
 ParserStatus Parser::parseLineDirective(bool isLine) {
+  SyntaxParsingContext PoundSourceLocation(SyntaxContext,
+                                           SyntaxKind::PoundSourceLocation);
```

Comment:
Good catch!
Could you separate `#sourceLocation` parsing out of this PR if possible?